pkg/pillar/cmd/upgradeconverter/upgradeconverter.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2017-2018 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package upgradeconverter
+
+import (
+	"flag"
+	"os"
+
+	"github.com/lf-edge/eve/pkg/pillar/agentlog"
+	"github.com/lf-edge/eve/pkg/pillar/pidfile"
+	"github.com/lf-edge/eve/pkg/pillar/pubsub"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	log "github.com/sirupsen/logrus"
+)
+
+// ConverterAction - Defines the types of action that can occur when moving between versions
+type ConverterAction uint8
+
+const (
+	// Convert - Convert from old to new
+	Convert ConverterAction = iota + 1
+	// CleanupOld - Delete old versions
+	CleanupOld
+	// CleanupNew - Delete new versions if outdated
+	CleanupNew
+)
+
+var conversionHandlers = []ConversionHandler{
```

Comment:
Ok, all this needs to be properly documented in terms of what is expected of these converter hooks @kalyan-nidumolu

Reply:
This is the case for handling conversion for different modules - There is no expectation for these do. They can generate / initialize new files OR convert old Config Or do anything.. Will add appropriate comments